cli/cli_app.go

```diff
@@ -335,6 +335,12 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) (finalStatus er
 		return false
 	}
 
+	if conf.Uniqueness != nil {
```

Comment:
This was moved earlier in the process to make sure that the temporary directory is based on the environment and region